src/main/kotlin/org/koil/image/AccountImageController.kt

```diff
@@ -0,0 +1,43 @@
+package org.koil.image
+
+import org.koil.auth.EnrichedUserDetails
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.multipart.MultipartFile
+import org.springframework.web.servlet.ModelAndView
+import java.net.URI
+
+
+@Controller
+@RequestMapping("/dashboard/user-settings/image")
+class AccountImageController(
+    private val profileImageService: ProfileImageService
+) {
+    @GetMapping
+    fun getImageUrl(@AuthenticationPrincipal user: EnrichedUserDetails): ResponseEntity<Unit> {
+        val image = profileImageService.getImageUrlForUser(user.accountId)
+
+        return ResponseEntity.status(HttpStatus.FOUND)
+            .location(URI(image))
+            .build()
+    }
+
+    @PostMapping
+    fun handleFileUpload(
+        @AuthenticationPrincipal user: EnrichedUserDetails,
+        @RequestParam("file") file: MultipartFile
+    ): ModelAndView {
+        profileImageService.saveImageForUser(
+            user.accountId,
+            file
+        )
+
+        return ModelAndView("redirect:/dashboard/user-settings?imageUploaded=true")
```

Comment:
I can't find that we use ?imageUploaded on the template.

Reply:
We don't right now. I'll add it in a future change, but wanted the API to be fixed as early as possible.